dmapiclient/search.py

```diff
@@ -98,9 +101,9 @@ def set_alias(self, alias_name, target_index):
             data={'type': 'alias', 'target': target_index}
         )
 
-    def index(self, index, service_id, service):
-        url = self._url(index, service_id)
-        return self._put(url, data={'service': service})
+    def index(self, index_name, object_type, object_id, serialized_object):
+        url = '/{}/{}/{}'.format(index_name, object_type, object_id)
+        return self._put(url, data={'document': serialized_object})
```

Comment:
Is there any value in defaulting the object_type to `service` for backwards compatibility (i.e. would this avoid the breaking change)?

Reply:
Yeah, i did think about it, but then the change from 'service' to 'document' in the data is a breaking change anyway.

Reply:
Hmm, annoyingly in this case 'object type' has to be `services` with an ess - is possibly pedantry to mention it, but maybe it means we can think of another variable name. It's called type elsewhere, though.

Still wondering if having a default here might be a good idea, to keep backward compatibility?

The service->document change isn't breaking I think, in the sense that it's entirely captured here (i.e. clients of the library don't need to know about it).

Reply:
@galund I'm a little confused about the '`services` with an ess' thing?

And I guess now that the search-api can handle 'document' or 'services' as the key it wouldn't actually break anything. 

I'll add the default and remove the breaking change.

Reply:
Get said "add 'service' for backward compat" and I was just pointing out that you have to add "service*s*" (unless I'm wrong!)

Reply:
I think it'd need to be `service`, as that's what the search-api is currently looking for, and also what you're using in your current PR on the search-api?

Reply:
Object type is being used in the URL - the "document" string is hard-coded (which is fine).

Reply:
Yep, sorry, I'm with you, you're right. It was `services` that was being hardcoded into the url before.